[web][api][SIMS #532]Institution User Edits Student Enrolment (CoE)

andrewsignori-aot opened the pull request:
- Added the call in the COE to "Edit the Program Information". The menu item was renamed after a conversation with @abschwenker 
![image](https://user-images.githubusercontent.com/61259237/131387000-3ddc5c6c-a419-4a8d-a5af-d7fe439005d7.png)

- Restrict API queries to do not consider Overwiritten applications while retrieving data from DB. Many queries on application service were changed.

- Added below modal dialog for confirmation of the CoE Edit
![image](https://user-images.githubusercontent.com/61259237/131387110-8da18574-af5d-4ba3-bb5f-db30a8237d73.png)

- Created a generic modal dialog based on Veutify and based on Promises instead of events (please let me know your thoughts about it).
The generic component has a type that allows us to introduce more features and also Vue slots that allow full customization of header, content, and footer, as shown below.

![image](https://user-images.githubusercontent.com/61259237/131387336-a35a8cf3-6582-4ab2-bd82-20103764912a.png)

![image](https://user-images.githubusercontent.com/61259237/131387352-5df87d00-7740-4647-a234-ea1549be7d18.png)

guru-aot commented on sources/packages/web/src/services/http/ConfirmationOfEnrollmentApi.ts:
```diff
@@ -46,4 +46,20 @@ export class ConfirmationOfEnrollmentApi extends HttpBaseClient {
+    applicationId: number,
+  ): Promise<void> {
+    try {
+      await this.apiClient.post(
```
is it possible to have a GET instead of a POST? it does not have a payload, so wanted to check, is there a reason to have it as post call?

guru-aot replied:
Cool i get it now, thanks

guru-aot reviewed (approved):
LGTM, other than the doubt on the POST call and the failing Test cases

andrewsignori-aot commented:
> LGTM, other than the doubt on the POST call and the failing Test cases

Needed some additional work on E2E tests. I just updated the PR with some modifications. I added a restriction to once the application is in some final state (Overwritten, Cancelled, Completed) it will no longer allow the change in the status and it caused some issues.

ann-aot reviewed (approved):
LGTM 👍 Liked the `ModalDialogBase` generic component with flexible slots
